service/frontend/namespace_handler.go

```diff
@@ -98,28 +99,35 @@ const (
 	maxReplicationHistorySize = 10
 )
 
-var ErrInvalidNamespaceStateUpdate = serviceerror.NewInvalidArgument("invalid namespace state update")
-var _ Handler = (*HandlerImpl)(nil)
+var _ NamespaceHandler = (*namespaceHandlerImpl)(nil)
 
-// NewHandler create a new namespace handler
-func NewHandler(
+var (
+	// err indicating that this cluster is not the master, so cannot do namespace registration or update
+	errNotMasterCluster                   = serviceerror.NewInvalidArgument("Cluster is not master cluster, cannot do namespace registration or namespace update.")
+	errCannotDoNamespaceFailoverAndUpdate = serviceerror.NewInvalidArgument("Cannot set active cluster to current cluster when other parameters are set.")
+	errInvalidRetentionPeriod             = serviceerror.NewInvalidArgument("A valid retention period is not set on request.")
+	errInvalidNamespaceStateUpdate        = serviceerror.NewInvalidArgument("Invalid namespace state update.")
+)
+
+// NewNamespaceHandler create a new namespace handler
+func NewNamespaceHandler(
 	maxBadBinaryCount dynamicconfig.IntPropertyFnWithNamespaceFilter,
 	logger log.Logger,
 	metadataMgr persistence.MetadataManager,
 	clusterMetadata cluster.Metadata,
-	namespaceReplicator Replicator,
+	namespaceReplicator namespace.Replicator,
 	archivalMetadata archiver.ArchivalMetadata,
 	archiverProvider provider.ArchiverProvider,
 	supportsSchedules dynamicconfig.BoolPropertyFnWithNamespaceFilter,
 	timeSource clock.TimeSource,
-) *HandlerImpl {
-	return &HandlerImpl{
+) *namespaceHandlerImpl {
```

Comment:
I think you need to fix this, and add `//nolint` comment for all the rest.

Reply:
Fixed this one, but gonna leave the others. This diff is a refactor, so I don't wanna silence a lint warning that might be useful later.